x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichCacheTests.java

```diff
@@ -89,13 +95,14 @@ public void testCaching() {
 
         var enrichCache = new EnrichCache(3);
         enrichCache.setMetadata(metadata);
-        enrichCache.put(searchRequest1, searchResponse);
-        enrichCache.put(searchRequest2, searchResponse);
-        enrichCache.put(searchRequest3, searchResponse);
+        // warming the cache means there's a miss
+        warmCache(enrichCache, searchRequest1, searchResponse);
+        warmCache(enrichCache, searchRequest2, searchResponse);
+        warmCache(enrichCache, searchRequest3, searchResponse);
         var cacheStats = enrichCache.getStats("_id");
         assertThat(cacheStats.getCount(), equalTo(3L));
         assertThat(cacheStats.getHits(), equalTo(0L));
-        assertThat(cacheStats.getMisses(), equalTo(0L));
+        assertThat(cacheStats.getMisses(), equalTo(3L));
```

Comment:
why is the expected miscount increased here?

Reply:
Without introducing a method specifically for the tests, warming the cache requires a cache miss to trigger a resolve.

The class is also final so can't do the usual trickery to break encapsulation. A specific put or constructor where we can pass in a `Cache<CacheKey, CompletableFuture<SearchResponse>> cache` are alternatives - I don't know whats preferred?

Reply:
I see. Maybe we should make `EnrichCache` non final?

Reply:
Adressed in https://github.com/elastic/elasticsearch/pull/77259/commits/14ef0a2eb130895d04086722545ca5c7ea1cc715